Content.Server/Anomaly/Effects/ChasingAnomalySystem.cs

```diff
@@ -0,0 +1,79 @@
+using System.Linq;
+using System.Numerics;
+using Content.Server.Anomaly.Components;
+using Content.Shared.Anomaly.Components;
+using Robust.Shared.Physics.Systems;
+using Robust.Shared.Random;
+
+namespace Content.Server.Anomaly.Effects;
+/// <summary>
+/// This component allows the anomaly to chase a random instance of the selected type component within a radius.
+/// </summary>
+public sealed class ChasingAnomalySystem : EntitySystem
+{
+    [Dependency] private readonly EntityLookupSystem _lookup = default!;
+    [Dependency] private readonly SharedPhysicsSystem _physics = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+
+    /// <inheritdoc/>
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<ChasingAnomalyComponent, AnomalyPulseEvent>(OnPulse);
+    }
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        foreach (var (anom, trans) in EntityManager.EntityQuery<ChasingAnomalyComponent, TransformComponent>(true))
+        {
+            if (!Initialized(anom.Owner)) return;
+            if (Deleted(anom.ChasingEntity)) continue;
+            if (!anom.ChasingEntity.IsValid()) continue;
```

Comment:
just do `|| anom.ChasingEntity == null` after its changed to be `EntityUid?`

also have this

```
if (!anom.ChasingEntity.Valid())
{
    anom.ChasingEntity = null;
    continue;
}
```

so if it gets deleted it can be a tiny bit faster when checking afterwards

Reply:
@deltanedas 
when changing to EntityUid? problems with functions, do you need any other options? Exactly: 
```
.Chasing Entity.Valid()
.GetComponent(anom.ChasingEntity);
```

Reply:
the first will work as long as you check `anom.ChasingEntity == null` before it
the second just do `anom.ChasingEntity.Value`